src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    public static final int OPERATION_TYPE = 0;
+    public static final int AMOUNT = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File fromFile = new File(fromFileName);
+        File toFile = new File(toFileName);
+        int supplySum = 0;
+        int buySum = 0;
+        int result = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                String[] words = value.split(",");
+                if (words[OPERATION_TYPE].equals("supply")) {
+                    supplySum += Integer.parseInt(words[AMOUNT]);
+                } else if (words[OPERATION_TYPE].equals("buy")) {
+                    buySum += Integer.parseInt(words[AMOUNT]);
+                }
+                value = bufferedReader.readLine();
+            }
+            result += (supplySum - buySum);
 
+            String report = "supply," + supplySum + System.lineSeparator()
```

Comment:
Use `StringBuilder` instead of `String` to concatenate data in loops or when building a multi-line string. This is more efficient, especially when dealing with loops or large amounts of data concatenation.